### ISSUE TITLE
Last character doesn't appear in Select when there is a special character

### ISSUE THREAD

**Issue by GaetanBaert** [fyne-io/fyne#4293]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

The last character of the option selected doesn't appear in widget.Select when the option selected contains a specific character like "à", "é"... 

### How to reproduce

I include a small piece of code to reproduce the bug.

### Screenshots

![bug_report](https://github.com/fyne-io/fyne/assets/47001815/b3250358-2c43-4d23-b05a-0568d284cfb5)

### Example code

```go
package main

import (
	"log"

	"fyne.io/fyne/v2"
	"fyne.io/fyne/v2/app"
	"fyne.io/fyne/v2/widget"
)

func main() {
	a := app.New()
	w := a.NewWindow("Test")
	w.Resize(fyne.NewSize(640, 480))
	options := []string{"Here is à test", "This is a test"}
	sel := widget.NewSelect(options, func(v string) {
		log.Println(v)
	})
	w.SetContent(sel)
	w.ShowAndRun()
}

```

### Fyne version

v2.4.0

### Go compiler version

1.21.0

### Operating system and version

Windows 11

**Comment by andydotxyz** [comment]
Please can you try the develop branch for Fyne? I think this was fixed already.

**Comment by GaetanBaert** [comment]
I'm sorry, I'm quite new to Go

How can I try the fix ? I need to clone the branch and call it from local ?

**Comment by andydotxyz** [comment]
Globe the repo with git and change to the right branch (GitHub shows instructions on the PR).
Then add a "replace" directive in go.mod to show your project to use a file location for fyne.io/fyne/v2 instead.

**Comment by GaetanBaert** [comment]
I tried, and I confirm the fix at #4286 solves the issue. Thank you !

**Comment by andydotxyz** [comment]
Excellent :) we should have that in a release in the coming days
